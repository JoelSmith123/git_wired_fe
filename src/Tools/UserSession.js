export default class UserSession {
  constructor() {
    this.tokenKey = 'userToken'
  }

  setGitWiredToken(token) {
    sessionStorage.setItem(this.tokenKey, token);
  }

  getGitWiredToken() {
    return sessionStorage.getItem(this.tokenKey);
  }

  logout() {
    sessionStorage.removeItem(this.tokenKey);
  }


  isLoggedIn = () => {
    return Boolean( this.getGitWiredToken() )
  }




  // Make Fetch /login call
  // receive userToken

  // TO DO - TEST ME
  //  - CONFIRM endpoint
  //  - CONFIRM RESPONSE STRUCTURE
  // loginUser() {
  //   let url = 'https://git-wired-be.herokuapp.com/login'
  //   fetch(url)
  //     .then(response => response.json())
  //     .then(data     => this.getToken(data).bind(this))
  //     .then(token    => this.setGitWiredToken(token).bind(this))
  //     .catch(error   => { console.log(error); });
  // }
  //
  // getToken(data) {
  //   let token = data[this.tokenKey]
  //   return token
  // }


}
