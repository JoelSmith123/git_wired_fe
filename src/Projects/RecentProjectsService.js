
import UserSession from '../tools/UserSession.js'


export default class RecentProjectsService {

  constructor() {
    this.user = new UserSession
  }

  stubRecent() {
    let stub =  {
      'data': {
        'id':         1,
        'attributes': {
          'recent': [
            {
              'projectTitle':   "Wireframe Title 1",
              'repoName':       "Repo 1",
              'updatedAt':      "Updated At",
              'createdAt':      "Created At",
            },
            {
              'projectTitle':   "Wireframe Title 2",
              'repoName':       "Repo 2",
              'updatedAt':      "Updated At",
              'createdAt':      "Created At",
            },
          ],
        }
      },
    }
    return stub
  }


  getRecents = (parseFunc) => {
    // TO DO - Confirm ENDPOINT
    // let url = 'https://git-wired-be.herokuapp.com/api/v1/projects'
    let url = 'https://git-wired-be.herokuapp.com/api/v1/wireframes'
    fetch(url, {
        method:  "GET",
        headers: { "Content-Type": "application/json", },
        body:    JSON.stringify(this.user.getGitWiredToken()),
    })
    .then(response => response.json() )
    .then(data     => parseFunc(data))
    .catch(error   => console.error('Error:', error));
  }



}
