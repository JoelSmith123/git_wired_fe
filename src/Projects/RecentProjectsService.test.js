import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import RecentProjectsService from './RecentProjectsService'
import UserSession from '../Tools/UserSession.js'

describe('RecentProjectsService', () => {
  let recentProjectsService
  let mockStub
  let user
  beforeEach(() => {
    recentProjectsService = new RecentProjectsService()
    user = new UserSession

    mockStub =  {
      'data': {
        'id':         1,
        'attributes': {
          'recent': [
            {
              'projectTitle':   "Wireframe Title 1",
              'repoName':       "Repo 1",
              'updatedAt':      "Updated At",
              'createdAt':      "Created At",
            },
            {
              'projectTitle':   "Wireframe Title 2",
              'repoName':       "Repo 2",
              'updatedAt':      "Updated At",
              'createdAt':      "Created At",
            },
          ],
        }
      },
    }

    window.fetch = jest.fn().mockImplementation(() => {
      Promise.resolve({ok: true, json: () => Promise.resolve(mockStub.recent)})})
  })

  it('should match snapshot', () => {
    const wrapper = recentProjectsService
    expect(wrapper).toMatchSnapshot()
  })    

  it('should return new stub on stubRecent invokation', () => {
    const wrapper = recentProjectsService
    expect(wrapper.stubRecent()).toEqual(mockStub)
  })

  it('should check if fetch is called with correct params in getRecents', () => {
    const wrapper = recentProjectsService
    const parseFunc = jest.fn()
    const expected = [
      'https://git-wired-be.herokuapp.com/api/v1/wireframes',
      {
        method:  "GET",
        headers: { "Content-Type": "application/json", },
        body:    JSON.stringify(user.getGitWiredToken()),
      } 
    ]

    wrapper.getRecents(parseFunc)
    expect(window.fetch).toHaveBeenCalledWith(...expected)
  })
})

