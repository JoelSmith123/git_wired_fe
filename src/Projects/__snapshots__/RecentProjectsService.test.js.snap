// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentProjectsService should match snapshot 1`] = `
RecentProjectsService {
  "getRecents": [Function],
  "user": UserSession {
    "getURLToken": [Function],
    "isLoggedIn": [Function],
    "login": [Function],
    "tokenKey": "userToken",
  },
}
`;
